src/main/kotlin/org/sd_jwt/SdJwt.kt

```diff
@@ -408,7 +387,10 @@ inline fun internalCreatePresentation(
             .issueTime(Date.from(Instant.now()))
             .claim("nonce", nonce)
             .build()
-        presentation += SEPARATOR + buildJWT(holderBindingJwtPayload, holderKey)
+
+        val keyBasedSdJwtSigner = holderKey?.let { KeyBasedSdJwtSigner(it) }
```

Comment:
Should we not also use a signer for the holder key? Or is this not a practical use case? But I think it would be important for holder keys stored in HSMs.

Reply:
I agree, having a signer interface for the holder JWK would also be useful for implementors using the sd-jwt library as the holder.

I changed it now by adding a method to the SdJwtSigner for getting the public JWK.